### PR TITLE
[WIP] Refactor sample tests configuration to reduce the efforts of adding samples.

### PR DESCRIPTION
**Pull request by numerology**
[WIP] Refactor sample tests configuration to reduce the efforts of adding samples.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1724)

### CODE REVIEW

**Comment by gaoning777** [comment]
Since this PR is currently WIP, I will review once the tag is deleted.

File: test/sample-test/run_test.sh [hunk 1]
```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
...
+
+################################################################################
+# Utility function to prepare and validate the results for sample test based on
+# ipynb.
+# Arguments:
+#   Test name.
+################################################################################
+check_notebook_result() {
+  jupyter nbconvert --to python $1.ipynb
+  pip3 install tensorflow==1.8.0
```
Comment (Ark-kun) [inline test/sample-test/run_test.sh]: We might conserve time is we integrate tensorflow in our test image as opposed to dynamically installing it.

File: test/sample-test/run_test.sh [hunk 2]
```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
...
+  ipython $1.py
```
Comment (Ark-kun) [inline test/sample-test/run_test.sh]: Does this python file require ipython or can it run using the common python?

File: test/sample-test/run_test.sh [hunk 3]
```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
...
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" kubeflow-training-classification.py
+preparation ${TEST_NAME}
+
+if [[ "${TEST_NAME}" == "kubeflow-training-classification" ]]; then
+  #TODO: convert the sed commands to sed -e 's|gcr.io/ml-pipeline/|gcr.io/
+  # ml-pipeline-test/' and tag replacement.
+  # Compile samples.
+
+  if [[ -n "${DATAFLOW_TFT_IMAGE}" ]]; then
+    sed -i "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+\
```
Comment (Ark-kun) [inline test/sample-test/run_test.sh]: It seems to me that the previous formatting looked more readable.

File: test/sample-test/run_test.sh [hunk 4]
```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
...
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_PREDICT_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tfdv:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFDV_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tfma:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFMA_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-tf-trainer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DNNTRAINER_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DEPLOYER_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" taxi-cab-classification-pipeline.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-roc:\([a-zA-Z0-9_.-]\)\+|${LOCAL_ROC_IMAGE}|g" taxi-cab-classification-pipeline.yaml
+    sed -i "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+\
```
Comment (Ark-kun) [inline test/sample-test/run_test.sh]: Is it possible to extract this as a standalone function?

**Comment by numerology** [comment]
Sorry. I realized this PR was not based on my personal fork. Will resubmit and address @Ark-kun 's comments. Thanks!

File: test/sample-test/run_test.sh (hunk 3 above)
Comment (numerology) [reply test/sample-test/run_test.sh]: Agree. Changed it back. Also breaklines tend to cause problems.

**Pull request closed without merging**